core/src/main/java/hudson/Functions.java

```diff
@@ -1051,22 +1051,18 @@ public static String getFooterURL() {
      * we needed this for {@link GlobalConfiguration}s are for backward compatibility.
      *
      * @param predicate
-     *      Filter the descriptors based on {@link GlobalConfigurationCategory}
+     *      Filter the descriptors based on this predicate
      * @since 1.494
      */
-    public static Collection<Descriptor> getSortedDescriptorsForGlobalConfig(Predicate<GlobalConfigurationCategory> predicate) {
+    public static Collection<Descriptor> getSortedDescriptorsForGlobalConfig(Predicate<Descriptor> predicate) {
```

Comment:
isn't this going to break callers that pass in something expectiing to be called with a `GlobalConfigurationCategory`?
I found a few calls in the jenkinsci org.
noteably the [credentials plugin](https://github.com/jenkinsci/credentials-plugin/blob/6d79efcb436c5d39f745ec54ae080137fac8dbf4/src/main/java/com/cloudbees/plugins/credentials/GlobalCredentialsConfiguration.java#L70-L74) [here](https://github.com/jenkinsci/credentials-plugin/blob/6d79efcb436c5d39f745ec54ae080137fac8dbf4/src/main/java/com/cloudbees/plugins/credentials/GlobalCredentialsConfiguration.java#L166)

Reply:
Oh no 😭

Reply:
very much broken 😢 , two references I can see, credentials and aws-global-configuration-plugin

Reply:
anyone got an idea on the best option?

could update both plugins, but it'd be a compatibility break 😢

Reply:
Can we introduce this as a new function and leave the old function alone but deprecate and restrict it?

Reply:
that was that I was thinking, there's ~4 functions, I need to check each one to see if all of them would need the same treatment,

(for background the change is required so we can get the descriptors required permission due to the manage permission)